File: Data-Agent-Server/data-agent-server-app/src/main/java/edu/zsc/ai/domain/service/db/DatabaseService.java
package edu.zsc.ai.domain.service.db;

import java.util.List;

public interface DatabaseService {

    List<String> listDatabases(Long connectionId);

    /**
     * List databases for a connection, with explicit user for ownership. When userId is null, uses current login (StpUtil).
     */
    List<String> listDatabases(Long connectionId, Long userId);

    void deleteDatabase(Long connectionId, String databaseName, Long userId);

    /**
     * Get list of available character sets for a connection
     * @param connectionId connection id
     * @return list of character set names
     */
    List<String> getCharacterSets(Long connectionId);

    /**
     * Get list of available collations for a given character set
     * @param connectionId connection id
     * @param charset character set name
     * @return list of collation names
     */
    List<String> getCollations(Long connectionId, String charset);

    /**
     * Create a new database
     * @param connectionId connection id
     * @param databaseName database name
     * @param charset character set
     * @param collation collation (sorting rule)
     * @param userId user id
     */
    void createDatabase(Long connectionId, String databaseName, String charset, String collation, Long userId);

    /**
     * Check if a database exists
     * @param connectionId connection id
     * @param databaseName database name to check
     * @param userId user id
     * @return true if database exists
     */
    boolean databaseExists(Long connectionId, String databaseName, Long userId);
}

File: Data-Agent-Client/src/services/database.service.ts
import http from '../lib/http';
import { ApiPaths } from '../constants/apiPaths';

/**
 * List database names on an active connection.
 * Requires the connection to be opened first via connectionService.openConnection.
 */
export const databaseService = {
  listDatabases: async (connectionId: string): Promise<string[]> => {
    const response = await http.get<string[]>(ApiPaths.DATABASES, {
      params: { connectionId },
    });
    return response.data;
  },

  deleteDatabase: async (connectionId: string, databaseName: string): Promise<void> => {
    await http.post(`${ApiPaths.DATABASES}/delete`, null, {
      params: { connectionId, databaseName },
    });
  },

  /**
   * Get list of available character sets for a connection
   */
  getCharacterSets: async (connectionId: string): Promise<string[]> => {
    const response = await http.get<string[]>(`${ApiPaths.DATABASES}/charsets`, {
      params: { connectionId },
    });
    return response.data;
  },

  /**
   * Get list of available collations for a given character set
   */
  getCollations: async (connectionId: string, charset: string): Promise<string[]> => {
    const response = await http.get<string[]>(`${ApiPaths.DATABASES}/collations`, {
      params: { connectionId, charset },
    });
    return response.data;
  },

  /**
   * Create a new database
   */
  createDatabase: async (
    connectionId: string,
    databaseName: string,
    charset: string,
    collation?: string
  ): Promise<void> => {
    await http.post(`${ApiPaths.DATABASES}/create`, null, {
      params: { connectionId, databaseName, charset, collation },
    });
  },

  /**
   * Check if a database exists
   */
  databaseExists: async (connectionId: string, databaseName: string): Promise<boolean> => {
    const response = await http.get<boolean>(`${ApiPaths.DATABASES}/exists`, {
      params: { connectionId, databaseName },
    });
    return response.data;
  },
};
